docs/standard/datetime/datetime-support-in-system-text-json.md

```diff
@@ -0,0 +1,199 @@
+
+## Custom support for <xref:System.DateTime> and <xref:System.DateTimeOffset> in `JsonSerializer`
+
+If you want the serializer to perform custom parsing or formatting, you can implement
+[custom converters](https://docs.microsoft.com/dotnet/api/system.text.json.serialization.jsonconverter-1?view=netcore-3.0).
+Here are a few examples:
+
+### Using `DateTime(Offset).Parse` and `DateTime(Offset).ToString`
+
+If you can't determine the formats of your input <xref:System.DateTime> or <xref:System.DateTimeOffset> text representations, you can use the
+`DateTime(Offset).Parse` method in your converter read logic. This allows you to use .NET's extensive support for parsing various
+<xref:System.DateTime> and <xref:System.DateTimeOffset> text formats, including non-ISO 8601 strings and ISO 8601 formats that don't conform to
+the extended ISO 8601-1:2019 profile. This approach is significantly less performant than using the serializer's native implementation.
+
+For serializing, you can use `DateTime(Offset).ToString` method in your converter write logic. This allows you to write <xref:System.DateTime>
+and <xref:System.DateTimeOffset> values using any of the
+[standard date and time formats](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings).
+This approach is also significantly less performant than using the serializer's native implementation.
+
+[!code-csharp[example-showing-datetime-parse](~/samples/snippets/standard/datetime/json/datetime-converter-examples/example1/Program.cs)]
+
+```markdown
+> [!NOTE]
+> When implementing JsonConverter<DateTime>, the `typeToConvert` parameter will always be typeof(DateTime).
+> The parameter is useful for polymorphic cases and when using generics to get typeof(T) in a performant way.
+```
+
+### Using <xref:System.Buffers.Text.Utf8Parser> and <xref:System.Buffers.Text.Utf8Formatter>
+
+You can used fast UTF-8-based parsing and formatting methods in your converter logic if your input <xref:System.DateTime> or <xref:System.DateTimeOffset>
+text representations are compliant with one of the "R", "l", "O", or "G"
+[standard date and time Format Strings](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings),
+or you want to write according to one of these formats. This is much faster than using `DateTime(Offset).Parse` and `DateTime(Offset).ToString
+
+This example shows a custom converter that serializes and deserializes <xref:System.DateTime> values according to
+[the "R" standard format](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings#the-rfc1123-r-r-format-specifier):
+
+[!code-csharp[example-showing-utf8-parser-and-formatter](~/samples/snippets/standard/datetime/json/datetime-converter-examples/example2/Program.cs)]
+
+```markdown
+> [!NOTE]
+> The "R" standard format will always be 29 characters long.
```

Comment:
It this the expected output format? I would have thought it would be formatted differently. Why add the `[!NOTE]` if it just renders as plaintext?

https://review.docs.microsoft.com/en-us/dotnet/standard/datetime/datetime-support-in-system-text-json?branch=pr-en-us-13526
![image](https://user-images.githubusercontent.com/6527137/62904514-8b478e80-bd1b-11e9-9cf8-e86ed4d8ba04.png)

Reply:
I suspect the markdown wrapper shouldn’t be there. I’ll check shortly.